src/main/scala/strawman/collection/View.scala

```diff
@@ -266,22 +264,102 @@ trait ArrayLike[+A] extends Any {
   def apply(i: Int): A
 }
 
+trait SeqView[+A] extends View[A] with Seq[A] with SeqOps[A, SeqView, SeqView[A]] { self =>
+  override protected[this] def coll: this.type = this
+
+  override def view: SeqView[A] = this
+
+  override def iterableFactory: SeqFactory[SeqView] = SeqView
+
+  override protected[this] def fromSpecificIterable(it: Iterable[A]): SeqView[A] =
+    it match {
+      case v: SeqView[A] => v
+      case i: Seq[A] => i.view
+      case _ => it.to(Seq).view
+    }
+
+  override protected[this] def newSpecificBuilder(): Builder[A, SeqView[A]] =
+    Seq.newBuilder[A]().mapResult(_.view)
+
+  override def apply(i: Int) = iterator().drop(i - 1).next()
+  override def prepended[B >: A](elem: B): SeqView[B] = SeqView.Prepend(elem, this)
+  override def appended[B >: A](elem: B): SeqView[B] = SeqView.Append(this, elem)
+  override def prependedAll[B >: A](prefix: Iterable[B]): SeqView[B] = SeqView.PrependAll(prefix, this)
+  override def appendedAll[B >: A](suffix: Iterable[B]): SeqView[B] = SeqView.AppendAll(this, suffix)
+}
+
+object SeqView extends SeqFactory[SeqView] {
+  def fromIterator[A](it: => Iterator[A]): SeqView[A] = new SeqView[A] {
+    override def length = it.size
+    override def apply(i: Int) = it.drop(i - 1).next()
+    def iterator() = it
+  }
+
+  /** Avoid copying if source collection is already a view. */
+  def fromIterable[E](it: Iterable[E]): SeqView[E] = it match {
+    case it: SeqView[E] => it
+    case _ => SeqView.fromIterator(it.iterator())
+  }
+
+  def empty[A]: SeqView[A] = Empty
+
+  /** The empty `SeqView` */
+  case object Empty extends SeqView[Nothing] {
+    override def length = 0
+    def iterator() = Iterator.empty
+    override def knownSize = 0
+  }
+
+  def newBuilder[A](): Builder[A, SeqView[A]] = ArrayBuffer.newBuilder[A]().mapResult(fromIterable)
+
+  /** A view that appends an element to its elements */
+  case class Append[A](underlying: Iterable[A] with ArrayLike[A], elem: A) extends SeqView[A] {
```

Comment:
I guess here we can not write `SeqOps[A, _, _]` because of the bounds on type parameters `CC[_]` and `C`?